ScreenCapture.NET/Model/GraphicsCard.cs

```diff
@@ -1,86 +1,85 @@
-        /// </summary>
-        /// <param name="index">The index of the <see cref="GraphicsCard"/>.</param>
-        /// <param name="name">The name of the <see cref="GraphicsCard"/>.</param>
-        /// <param name="vendorId">The vendor-id of the <see cref="GraphicsCard"/>.</param>
-        /// <param name="deviceId">The device-id of the <see cref="GraphicsCard"/>.</param>
-        public GraphicsCard(int index, string name, int vendorId, int deviceId)
-        {
-            this.Index = index;
-            this.Name = name;
-            this.VendorId = vendorId;
-            this.DeviceId = deviceId;
-        }
-
-        #endregion
-
-        #region Methods
-
-        /// <summary>
-        /// Determines whether this <see cref="Display"/> equals the given one.
-        /// </summary>
-        /// <param name="other">The <see cref="Display"/> to compare.</param>
-        /// <returns><c>true</c> if the specified object is equal to the current object; otherwise, <c>false</c>.</returns>
-        public bool Equals(GraphicsCard other) => Index == other.Index;
-
-        /// <inheritdoc />
-        public override bool Equals(object? obj) => obj is GraphicsCard other && Equals(other);
-
-        /// <inheritdoc />
-        public override int GetHashCode() => Index;
-
-        /// <summary>
-        /// Determines whether two <see cref="GraphicsCard"/> are equal.
-        /// </summary>
-        /// <param name="left">The first value.</param>
-        /// <param name="right">The second value.</param>
-        /// <returns><c>true</c> if the two specified graphics-cards are equal; otherwise, <c>false</c>.</returns>
-        public static bool operator ==(GraphicsCard left, GraphicsCard right) => left.Equals(right);
-
-        /// <summary>
-        /// Determines whether two <see cref="GraphicsCard"/> are not equal.
-        /// </summary>
-        /// <param name="left">The first value.</param>
-        /// <param name="right">The second value.</param>
-        /// <returns><c>true</c> if the two specified graphics-cards are not equal; otherwise, <c>false</c>.</returns>
-        public static bool operator !=(GraphicsCard left, GraphicsCard right) => !(left == right);
-
-        #endregion
+        this.Index = index;
+        this.Name = name;
+        this.VendorId = vendorId;
+        this.DeviceId = deviceId;
     }
-}
+
+    #endregion
+
+    #region Methods
+
+    /// <summary>
+    /// Determines whether this <see cref="Display"/> equals the given one.
+    /// </summary>
+    /// <param name="other">The <see cref="Display"/> to compare.</param>
+    /// <returns><c>true</c> if the specified object is equal to the current object; otherwise, <c>false</c>.</returns>
+    public bool Equals(GraphicsCard other) => Index == other.Index;
+
+    /// <inheritdoc />
+    public override bool Equals(object? obj) => obj is GraphicsCard other && Equals(other);
+
+    /// <inheritdoc />
+    public override int GetHashCode() => Index;
```

Comment:
For the same reasons as with the Display object, which is not much as based on ordering or anything else, since on most computers the graphics cards won't be changing as much in order to be an issue, but who knows, so I decided to make the following changes
```suggestion
    public bool Equals(GraphicsCard other) => Index == other.Index && VendorId == other.VendorId && DeviceId == other.DeviceId;

    /// <inheritdoc />
    public override bool Equals(object? obj) => obj is GraphicsCard other && Equals(other);

    /// <inheritdoc />
    public override int GetHashCode() => (Index, VendorId, DeviceId).GetHashCode();
```

Didn't include the name again as well, since I don't see a need to do so, since the VendorId and the DeviceId should always be different. In fact only the DeviceId should be enough, but, another int comparison has negligible impact considering the possible benefits.

Reply:
I absolutely agree with the change to the Display: The display index is unique for each graphics-card, so it was just wrong until now. But I don't see why the combination of index and vendor id would make sense here. Isn't it either index or vendorId +deviceid?

Reply:
Yeah, thinking about it, I just put more on because of uncertainty. Having the index only on the graphics card, makes sense, because how often will a card change the order in which it's installed? And then again with the VendorId and DeviceId, these are guaranteed to not change. But, I guess it makes sense on weird cases. A person who has 2 equal graphics cards, connected on SLI, the VendorId and DeviceId would be equal on them, since they just describe the card, and are set by the manufacturer. Also, if someone serializes the data, and then loads it afterwards, it still assumes that the index is the same, when it could have changed order. Plus the fact that depending on the library used, it could not retrieve in a deterministic way, who knows.

Reply:
Oh yeah, the index only would have issues with empty objects, and having the three of them, since they are used on the display comparison, makes it more robust on the display side as well.

In the end it's your library, I'm just throwing ideas out XD.

Reply:
Hmm I'd say the question here is mainly what kind of inconvenience is less of a probem.
Comparing all of them invalidates everything if the order really changes, comparing only the index would still work (for stored data for example) but maybe with an unexpected result since the index <-> name pair might have changed. But I think the later is better in the current situation since it worked like this until now and I don't really see that specific problem to be that realistic. I don't think the order of graphic cards will not change at all (I also don't think it will happen for displays without physically changing ports) anyway.

Reply:
Pretty much, do as you think would be best.